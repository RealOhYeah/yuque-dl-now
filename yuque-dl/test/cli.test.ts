import path from 'node:path'
import fs from 'node:fs'
import { fileURLToPath } from 'node:url'
import { afterEach, beforeEach, describe, expect, it } from 'vitest'
import { TestTools } from './helpers/TestTools'
import { KNOWLEDGE_BASE_URL } from './helpers/constant'

const __dirname = path.dirname(fileURLToPath(import.meta.url))
const cliPath = path.join(__dirname, '../bin/index.js')

let testTools: TestTools

const mdImgReg = /!\[.*?\]\(https*.*?\)/g

describe('yuque-dl CLI', () => {

  beforeEach(() => {
    testTools = new TestTools()
  })

  afterEach(() => {
    testTools.cleanup()
  })

  it('should work', async () => {
    const { stdout, exitCode, stderr } = await testTools.fork(cliPath, [
      KNOWLEDGE_BASE_URL.NORMAL,
      '-d', '.'
    ])
    expect(exitCode).toBe(0)
    expect(stdout).toContain('√ 已完成')
    const imgDir = path.join(testTools.cwd, '语雀公益计划/语雀·大学生公益计划/img')
    const indexMdPath = path.join(testTools.cwd, '语雀公益计划/语雀·大学生公益计划/index.md')
    expect(fs.existsSync(imgDir)).toBeTruthy()
    const data = fs.readFileSync(indexMdPath).toString()
    expect(data.match(mdImgReg)).toBeFalsy()
    expect(stderr).toBeFalsy()
  })

  it('ignore img should work ', async () => {
    const { stdout, exitCode } = await testTools.fork(cliPath, [
      KNOWLEDGE_BASE_URL.NORMAL,
      '-d', '.',
      '-i'
    ])
    expect(exitCode).toBe(0)
    expect(stdout).toContain('√ 已完成')
    const imgDir = path.join(testTools.cwd, '语雀公益计划/语雀·大学生公益计划/img')
    expect(fs.existsSync(imgDir)).toBeFalsy()
    const indexMdPath = path.join(testTools.cwd, '语雀公益计划/语雀·大学生公益计划/index.md')
    const data = fs.readFileSync(indexMdPath).toString()
    expect(data).toMatchSnapshot()
  })

  it('generate toc list should work ', async () => {
    const { stdout, exitCode } = await testTools.fork(cliPath, [
      KNOWLEDGE_BASE_URL.NORMAL,
      '-d', '.',
      '--toc',
      '-i'
    ])
    expect(exitCode).toBe(0)
    expect(stdout).toContain('√ 已完成')
    const imgDir = path.join(testTools.cwd, '语雀公益计划/语雀·大学生公益计划/img')
    expect(fs.existsSync(imgDir)).toBeFalsy()
    const indexMdPath = path.join(testTools.cwd, '语雀公益计划/语雀·大学生公益计划/index.md')
    const data = fs.readFileSync(indexMdPath).toString()
    expect(data).toMatchSnapshot()
  })
})
