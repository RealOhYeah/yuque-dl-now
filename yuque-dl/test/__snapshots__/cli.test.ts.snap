// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`yuque-dl CLI > generate toc list should work  1`] = `
"# 🎁 语雀 · 大学生公益计划

- [👉🏻 常见问题 👈🏻](#-%F0%9F%91%89%F0%9F%8F%BB-%E5%B8%B8%E8%A7%81%E9%97%AE%E9%A2%98-%F0%9F%91%88%F0%9F%8F%BB-)
- [公益计划介绍](#%E5%85%AC%E7%9B%8A%E8%AE%A1%E5%88%92%E4%BB%8B%E7%BB%8D)
  * [规则](#%E8%A7%84%E5%88%99)
  * [操作指引](#%E6%93%8D%E4%BD%9C%E6%8C%87%E5%BC%95)
    + [适合新用户：语雀官网公益计划页认证](#%E9%80%82%E5%90%88%E6%96%B0%E7%94%A8%E6%88%B7%E8%AF%AD%E9%9B%80%E5%AE%98%E7%BD%91%E5%85%AC%E7%9B%8A%E8%AE%A1%E5%88%92%E9%A1%B5%E8%AE%A4%E8%AF%81)
    + [适合老用户：会员信息页直接认证](#%E9%80%82%E5%90%88%E8%80%81%E7%94%A8%E6%88%B7%E4%BC%9A%E5%91%98%E4%BF%A1%E6%81%AF%E9%A1%B5%E7%9B%B4%E6%8E%A5%E8%AE%A4%E8%AF%81)

---

## <font style="background-color:#E8F7CF;"> </font><font style="background-color:#E8F7CF;">👉🏻</font><font style="background-color:#E8F7CF;"> 常见问题 </font><font style="background-color:#E8F7CF;">👈🏻</font><font style="background-color:#E8F7CF;"> </font>
[认证常见问题解决](https://www.yuque.com/yuque/welfare/faq)



## 公益计划介绍
助力梦想，相伴成长。

即日起，语雀面向**中国大陆各大学的大学生、教师**发布语雀公益计划-个人版。认证教育邮箱，即可免费享有语雀会员权益。



[语雀写给大学生的一封信](https://www.yuque.com/yuque/blog/welfare-edu)

### 规则
认证学校提供的 **edu.cn 结尾的教育邮箱**，即可免费获得 **1 年语雀会员**。

+ 若你之前并非会员/已是专业会员，则获得 1 年专业会员；
+ 若你已是超级会员，则获得 1 年超级会员。
+ 1 年后，可在原入口进行**续期**，只要你还在校期间，就可以一直免费使用。



### 操作指引
#### 适合新用户：语雀官网公益计划页认证
+ 打开 [语雀官网公益计划页](https://www.yuque.com/about/welfare) 

![](https://cdn.nlark.com/yuque/0/2023/png/103125/1694522534230-9d9d42f9-bbba-46d3-8e9a-773f2951089d.png)

+ 在「个人版公益计划」下，点击【**立即认证**】

![](https://cdn.nlark.com/yuque/0/2023/png/103125/1694522561681-3a568a52-7ef3-43ae-9067-d852f5556cf7.png)

+ 完成注册或登录后，来到「**账户设置-会员信息**」页面，输入教育邮箱，点击发送认证邮件

![](https://cdn.nlark.com/yuque/0/2023/png/103125/1694513696571-0458a2d9-9dd8-4eab-a04b-671e4fd66e09.png)

+ 在教育邮箱中查收该邮件，点击「**立即验证**」

![](https://cdn.nlark.com/yuque/0/2023/png/103125/1694513756376-8dc9a3bd-ec24-43b6-9a2f-5f7760fc421b.png)

+ 完成认证，直接获得会员 ✅

#### 适合老用户：会员信息页直接认证
+ 来到「账户设置-会员信息」页面，点击【**立即认证**】

![](https://cdn.nlark.com/yuque/0/2023/png/103125/1694511270833-fc586c50-cc11-45ef-8fae-96961ec92150.png)

![](https://cdn.nlark.com/yuque/0/2023/png/103125/1694513621870-1e0659bf-0bd5-4f55-9519-a9dc663f5111.png)



+ 输入教育邮箱，点击发送认证邮件

![](https://cdn.nlark.com/yuque/0/2023/png/103125/1694513696571-0458a2d9-9dd8-4eab-a04b-671e4fd66e09.png)

+ 在教育邮箱中查收该邮件，点击「立即验证」

![](https://cdn.nlark.com/yuque/0/2023/png/103125/1694513756376-8dc9a3bd-ec24-43b6-9a2f-5f7760fc421b.png)

+ 完成认证，直接获得会员 ✅







> 更新: 2023-10-07 14:12:28  
> 原文: <https://www.yuque.com/yuque/welfare/edu>"
`;

exports[`yuque-dl CLI > ignore img should work  1`] = `
"# 🎁 语雀 · 大学生公益计划

## <font style="background-color:#E8F7CF;"> </font><font style="background-color:#E8F7CF;">👉🏻</font><font style="background-color:#E8F7CF;"> 常见问题 </font><font style="background-color:#E8F7CF;">👈🏻</font><font style="background-color:#E8F7CF;"> </font>
[认证常见问题解决](https://www.yuque.com/yuque/welfare/faq)



## 公益计划介绍
助力梦想，相伴成长。

即日起，语雀面向**中国大陆各大学的大学生、教师**发布语雀公益计划-个人版。认证教育邮箱，即可免费享有语雀会员权益。



[语雀写给大学生的一封信](https://www.yuque.com/yuque/blog/welfare-edu)

### 规则
认证学校提供的 **edu.cn 结尾的教育邮箱**，即可免费获得 **1 年语雀会员**。

+ 若你之前并非会员/已是专业会员，则获得 1 年专业会员；
+ 若你已是超级会员，则获得 1 年超级会员。
+ 1 年后，可在原入口进行**续期**，只要你还在校期间，就可以一直免费使用。



### 操作指引
#### 适合新用户：语雀官网公益计划页认证
+ 打开 [语雀官网公益计划页](https://www.yuque.com/about/welfare) 

![](https://cdn.nlark.com/yuque/0/2023/png/103125/1694522534230-9d9d42f9-bbba-46d3-8e9a-773f2951089d.png)

+ 在「个人版公益计划」下，点击【**立即认证**】

![](https://cdn.nlark.com/yuque/0/2023/png/103125/1694522561681-3a568a52-7ef3-43ae-9067-d852f5556cf7.png)

+ 完成注册或登录后，来到「**账户设置-会员信息**」页面，输入教育邮箱，点击发送认证邮件

![](https://cdn.nlark.com/yuque/0/2023/png/103125/1694513696571-0458a2d9-9dd8-4eab-a04b-671e4fd66e09.png)

+ 在教育邮箱中查收该邮件，点击「**立即验证**」

![](https://cdn.nlark.com/yuque/0/2023/png/103125/1694513756376-8dc9a3bd-ec24-43b6-9a2f-5f7760fc421b.png)

+ 完成认证，直接获得会员 ✅

#### 适合老用户：会员信息页直接认证
+ 来到「账户设置-会员信息」页面，点击【**立即认证**】

![](https://cdn.nlark.com/yuque/0/2023/png/103125/1694511270833-fc586c50-cc11-45ef-8fae-96961ec92150.png)

![](https://cdn.nlark.com/yuque/0/2023/png/103125/1694513621870-1e0659bf-0bd5-4f55-9519-a9dc663f5111.png)



+ 输入教育邮箱，点击发送认证邮件

![](https://cdn.nlark.com/yuque/0/2023/png/103125/1694513696571-0458a2d9-9dd8-4eab-a04b-671e4fd66e09.png)

+ 在教育邮箱中查收该邮件，点击「立即验证」

![](https://cdn.nlark.com/yuque/0/2023/png/103125/1694513756376-8dc9a3bd-ec24-43b6-9a2f-5f7760fc421b.png)

+ 完成认证，直接获得会员 ✅







> 更新: 2023-10-07 14:12:28  
> 原文: <https://www.yuque.com/yuque/welfare/edu>"
`;
